lib/mocha.js

```diff
@@ -277,15 +283,67 @@ Mocha.prototype.invert = function() {
  * @param {Boolean} ignore
  * @return {Mocha}
  * @api public
- * @param {boolean} ignore
- * @return {Mocha}
  */
 Mocha.prototype.ignoreLeaks = function(ignore) {
   this.options.ignoreLeaks = Boolean(ignore);
   return this;
 };
 
 /**
+ * Enable or disable randomization of test execution order within suites.
+ *
+ * @param {(boolean|number|string)} [seed] Optional random seed.  Seed must be
+ *   a 32-bit unsigned integer, or a string which can convert to one.  If
+ *   `true`, a seed will be created.  If `false`, randomization will be
+ *   globally disabled (if it was enabled previously).
+ * @return {Mocha}
+ * @api public
+ */
+Mocha.prototype.randomize = function randomize(seed) {
+  if (seed !== false) {
+    if (!arguments.length || seed === true) {
+      // generate a number for seeding
+      seed = Mocha.seed();
+    } else if (!(typeof seed === 'number' && seed > 0
+      && seed <= MAX_SAFE_INTEGER && seed % 1 === 0)) {
+      // this would be a hexadecimal number represented as a string
+      if (/^(0[xX])?[a-fA-F0-9]{1,14}$'/.test(seed)) {
+        seed = parseInt(seed, 16);
+      } else {
+        // a decimal number as string, presumably
+        seed = parseInt(seed, 10);
+        if (isNaN(seed)) {
+          throw new Error('Invalid random seed. Expected unsigned 32-bit '
+            + 'integer; got: "' + seed + '"');
+        }
+      }
+    }
+
+    var engine = Random.engines.mt19937().seed(seed);
```

Comment:
Is there any option to, say, get a default randomization engine? It's nice to use a library with better randomization quality guaranteed than the browser-variable-quality `Math.random`, and it's not like we need something like cryptographic quality, but it would be cool to be able to pick up improvements to what's considered the average random number generator without having to go into the code and change which one is hardcoded. (For instance, I have heard talk of newer engines that manage to give better quality than the Mersenne Twister *while also being faster*. I don't know if such things are available in common use yet, but it would be neat to pick them up when they are.)